### ISSUE TITLE
[Bug]: VoiceOver does not announce the label of a Dropdown

### ISSUE THREAD

**Issue by UniqueClone** [microsoft/fluentui#27863]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (8) x64 Apple M1
    Memory: 34.27 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Browsers:
    Firefox: 105.0.1
    Safari: 16.4
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://developer.microsoft.com/en-us/fluentui#/controls/web/dropdown

## Actual Behavior
VoiceOver announces the selected option/placeholder text when focus lands on a dropdown.

## Expected Behavior
VoiceOver should announce the label and selected option/placeholder text when focus lands on a dropdown.

### Requested priority

High

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
